.github/workflows/build-publish.yml

```diff
@@ -10,31 +10,27 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@master
+    - name: Checkout code 🛒
+      uses: actions/checkout@v4
+
     - name: Update version file ⬆️
       uses: brettdorrans/write-version-to-file@v1.1.0
       with:
         filename: '.VERSION'
         placeholder: '${VERSION}'
 
-    - name: Set up Python 3.12
+    - name: Set up Python 3.12 🐍
       uses: actions/setup-python@v5
       with:
         python-version: "3.12"
 
-    - name: Install pypa/build
-      run: >-
-        python -m
-        pip install
-        build
-        --user
-
-    - name: Build a source tarball
-      run: >-
-        python -m
-        build
-        --sdist
-        --outdir dist/ .
+    - name: Install and configure Poetry 📜
+      uses: snok/install-poetry@v1
+      with:
+        virtualenvs-create: false
+
+    - name: Build & Package project 👷
+      run: poetry build
 
     - name: Publish distribution 📦 to PyPI
       if: startsWith(github.ref, 'refs/tags')
```

Comment:
do we need `poetry publish` here? 
Also do wee need to set new environent variables in github action?